neurokit2/signal/signal_fixpeaks.py

```diff
@@ -373,8 +373,13 @@ def _correct_misaligned(misaligned_idcs, peaks):
     valid_idcs = np.logical_and(misaligned_idcs > 1,
                                 misaligned_idcs < len(corrected_peaks))
     misaligned_idcs = misaligned_idcs[valid_idcs]
-    prev_peaks = corrected_peaks[[i - 1 for i in misaligned_idcs]]
-    next_peaks = corrected_peaks[[i + 1 for i in misaligned_idcs]]
+    prev_peaks = corrected_peaks[[i - 1 for i in misaligned_idcs if i - 1 > 0]]
```

Comment:
@zen-juen, note that the conditions that you are evaluating in the list comprehensions are already evaluated in `valid_idcs`.

Reply:
@JanCBrammer I made the changes because using actual data next_peaks was not computable because i+1 exceeded the index limit for misaligned_idcs

let me check again!

Reply:
You're right. Simply change `len(corrected_peaks)` to `len(corrected_peaks) - 1` in `valid_idcs`. Then you can remove the rest of the changes. Nice catch!

Reply:
I also realize that we can simplify https://github.com/neuropsychology/NeuroKit/blob/3f5efda8cd98e35c71aaa44cd80afb6dda87182d/neurokit2/signal/signal_fixpeaks.py#L354-L355

to `valid_idcs = missed_idcs > 1` since the indices are not incremented in `_correct_missed()`